std-bits/table/src/main/java/org/enso/table/data/column/operation/RequiresNumberFormatting.java

```diff
@@ -0,0 +1,44 @@
+package org.enso.table.data.column.operation;
+
+import org.enso.table.data.column.storage.ColumnStorage;
+import org.enso.table.data.column.storage.MixedStorage;
+import org.enso.table.data.table.Column;
+import org.graalvm.polyglot.Context;
+
+public class RequiresNumberFormatting {
+  /** Indicates whether a  column contains numbers greater than 1000000, and require formatting in viz*/
+  public static boolean apply(Column column) throws InterruptedException {
+    ColumnStorage storage = column.getStorage();
+    return applyToStorage(storage);
+  }
+
+  /**
+   * Indicates whether a column contains numbers greater than 1000000
+   *
+   * @return true/false if the column contains large numbers 
+   */
+  public static boolean applyToStorage(ColumnStorage storage) throws InterruptedException {
+    return (storage instanceof MixedStorage mixedStorage)
+        ? mixedStorage.cachedNumericFormatCheck()
+        : (boolean) compute(storage, Context.getCurrent());
+  }
+
+  /** Internal method performing the calculation on a storage. */
+  public static boolean compute(ColumnStorage storage, Context context) {
+    long size = storage.getSize();
+
+    long count = 0;
+    for (long i = 0; i < storage.getSize(); i++) {
+      var val = storage.getItemBoxed(i);
+      if (val instanceof Long n && (n > 999999 || n < -999999)) {
+        return true;
+      }
+
+      if (context != null) {
+        context.safepoint();
+      }
+    }
+
+    return false;
```

Comment:
I think we should have a short circuit here at the top.

If `storage.getType` is not numeric nor mixed, then we can immediately return `false` as it will surely not return a `Long` from `getItemBoxed` - so there is no point in looping through all the elements in e.g. a Text column.